core/services/ocr2/plugins/ocr2keeper/custom_telemetry.go

```diff
@@ -0,0 +1,161 @@
+package ocr2keeper
+
+import (
+	"context"
+	"encoding/hex"
+	"time"
+
+	"cosmossdk.io/errors"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/keeper_registry_wrapper2_0"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	evm21 "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/ocr2keeper/evm21"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/v2/core/static"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+type AutomationCustomTelemetryService struct {
+	utils.StartStopOnce
+	monitoringEndpoint commontypes.MonitoringEndpoint
+	blockSubscriber    *evm21.BlockSubscriber
+	blockSubChanID     int
+	threadCtrl         utils.ThreadControl
+	lggr               logger.Logger
+	configDigest       [32]byte
+	latestConfigDigest latestConfigDigestGetter
+}
+
+type latestConfigDigestGetter interface {
+	LatestConfigDetails(opts *bind.CallOpts) (keeper_registry_wrapper2_0.LatestConfigDetails, error)
+}
+
+// NewAutomationCustomTelemetryService creates a telemetry service for new blocks and node version
+func NewAutomationCustomTelemetryService(me commontypes.MonitoringEndpoint,
+	lggr logger.Logger, chain evm.Chain, rAddr common.Address, blocksub *evm21.BlockSubscriber) (*AutomationCustomTelemetryService, error) {
+	registry, rErr := keeper_registry_wrapper2_0.NewKeeperRegistry(rAddr, chain.Client())
+	if rErr != nil {
+		return nil, errors.Wrap(rErr, "error creating new Registry Wrapper for customTelemService")
+	}
+	return &AutomationCustomTelemetryService{
+		monitoringEndpoint: me,
+		threadCtrl:         utils.NewThreadControl(),
+		lggr:               lggr.Named("Automation Custom Telem"),
```

Comment:
looking at other parts of the codebase , this should be camelCase without spaces `AutomationCustomTelemetry`